bleak/backends/winrt/scanner.py

```diff
@@ -0,0 +1,194 @@
+import asyncio
+import logging
+import pathlib
+from typing import Callable, List
+from uuid import UUID
+
+from bleak.backends.device import BLEDevice
+from bleak.backends.scanner import BaseBleakScanner, AdvertisementData
+
+import winrt.windows.devices.bluetooth.advertisement as winbleadv
+import winrt.windows.storage.streams as winstrm
+
+
+logger = logging.getLogger(__name__)
+_here = pathlib.Path(__file__).parent
+
+
+def _format_bdaddr(a):
+    return ":".join("{:02X}".format(x) for x in a.to_bytes(6, byteorder="big"))
+
+
+def _format_event_args(e):
+    try:
+        return "{0}: {1}".format(
+            _format_bdaddr(e.bluetooth_address), e.advertisement.local_name or "Unknown"
+        )
+    except Exception:
+        return e.bluetooth_address
+
+
+class BleakScannerWinRT(BaseBleakScanner):
+    """The native Windows Bleak BLE Scanner.
+
+    Implemented using `Python/WinRT <https://github.com/Microsoft/xlang/tree/master/src/package/pywinrt/projection/>`_.
+
+    Keyword Args:
+        scanning mode (str): Set to "Passive" to avoid the "Active" scanning mode.
+
+    """
+
+    def __init__(self, **kwargs):
+        super(BleakScannerWinRT, self).__init__(**kwargs)
+
+        self.watcher = None
+        self._devices = {}
+        self._scan_responses = {}
+
+        if "scanning_mode" in kwargs and kwargs["scanning_mode"].lower() == "passive":
+            self._scanning_mode = winbleadv.BluetoothLEScanningMode.PASSIVE
+        else:
+            self._scanning_mode = winbleadv.BluetoothLEScanningMode.ACTIVE
+
+        self._signal_strength_filter = kwargs.get("SignalStrengthFilter", None)
+        self._advertisement_filter = kwargs.get("AdvertisementFilter", None)
+
+        self._received_token = None
+        self._stopped_token = None
+
+    def _received_handler(
+        self,
+        sender: winbleadv.BluetoothLEAdvertisementWatcher,
+        event_args: winbleadv.BluetoothLEAdvertisementReceivedEventArgs,
+    ):
+        """Callback for AdvertisementWatcher.Received"""
+        # TODO: Cannot check for if sender == self.watcher in winrt?
```

Comment:
How about `sender.equals(self.watcher)`? Although when would it never be not equal?

Reply:
Yes, that might work; did not think of that. In winrt you get a different memory address for `sender`, so I just let it slide. I *think* this will be equal all the time anyway...

Reply:
No, it did not work. I will leave it as it is. Maybe it works due to the wrong reasons in the `dotnet` backend...